File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        int n = nums.size();
        vector<int>ans;
        vector<int>freq(n+1,0);
        for(int num : nums){
            freq[num]++;
        }
        for(int i = 1; i<=n; i++){
            if(freq[i]>1){
                ans.push_back(i);
            }
        }
        return ans;
        
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        int n = nums.length;
        List<Integer> ans = new ArrayList<>();
        int[] freq = new int[n + 1];

        // Count the frequency of each number
        for (int num : nums) {
            freq[num]++;
        }

        // Add numbers that appear more than once
        for (int i = 1; i <= n; i++) {
            if (freq[i] > 1) {
                ans.add(i);
            }
        }

        return ans;
    }
}